javascript/npm/draco3d/draco_nodejs_example.js

```diff
@@ -82,81 +83,36 @@ function encodeMeshToFile(mesh, decoder) {
-  if (normalAttId > -1) {
-    console.log("Adding normal attribute.");
-    const norAttribute = decoder.GetAttribute(mesh, normalAttId);
-    const norAttributeData = new decoderModule.DracoFloat32Array();
-    decoder.GetAttributeFloatForAllPoints(mesh, norAttribute,
-        norAttributeData);
-    const normals = new Float32Array(numVertexCoord);
-    for (let i = 0; i < numVertexCoord; i += 3) {
-      normals[i] = norAttributeData.GetValue(i);
-      normals[i + 1] = norAttributeData.GetValue(i + 1);
-      normals[i + 2] = norAttributeData.GetValue(i + 2);
-    }
-    decoderModule.destroy(norAttributeData);
-    meshBuilder.AddFloatAttributeToMesh(newMesh, encoderModule.NORMAL,
-        numPoints, 3, normals);
-  }
+  const attrs = ["POSITION", "NORMAL", "TEX_COORD", "COLOR"];
 
-  const texAttId = decoder.GetAttributeId(mesh, decoderModule.TEX_COORD);
-  if (texAttId > -1) {
-    const texAttribute = decoder.GetAttribute(mesh, texAttId);
-    const texAttributeData = new decoderModule.DracoFloat32Array();
-    decoder.GetAttributeFloatForAllPoints(mesh, texAttribute,
-        texAttributeData);
-    assertEquals(numVertexCoord, texAttributeData.size());
-    const texcoords = new Float32Array(numVertexCoord);
-    for (let i = 0; i < numVertexCoord; i += 3) {
-      texcoords[i] = texAttributeData.GetValue(i);
-      texcoords[i + 1] = texAttributeData.GetValue(i + 1);
-      texcoords[i + 2] = texAttributeData.GetValue(i + 2);
+  attrs.forEach((attr) => {
+    const decoderAttr = decoderModule[attr];
+    const encoderAttr = decoderModule[attr];
+    const attrId = decoder.GetAttributeId(mesh, decoderAttr);
+
+    if (attrId < 0) {
+      return;
     }
-    decoderModule.destroy(texAttributeData);
-    meshBuilder.AddFloatAttributeToMesh(newMesh, encoderModule.TEX_COORD,
-        numPoints, 3, normals);
-  }
 
-  const colorAttId = decoder.GetAttributeId(mesh, decoderModule.COLOR);
-  if (colorAttId > -1) {
-    const colAttribute = decoder.GetAttribute(mesh, colorAttId);
-    const colAttributeData = new decoderModule.DracoFloat32Array();
-    decoder.GetAttributeFloatForAllPoints(mesh, colAttribute,
-        colAttributeData);
-    assertEquals(numVertexCoord, colAttributeData.size());
-    const colors = new Float32Array(numVertexCoord);
+    console.log("Adding %s attribute", attr);
+
+    const attribute = decoder.GetAttribute(mesh, attrId);
+    const attributeData = new decoderModule.DracoFloat32Array();
+    decoder.GetAttributeFloatForAllPoints(mesh, attribute, attributeData);
+
+    assert(numVertexCoord === attributeData.size(), 'Wrong attribute size.');
+
+    const attributeDataArray = new Float32Array(numVertexCoord);
     for (let i = 0; i < numVertexCoord; i += 3) {
-      colors[i] = colAttributeData.GetValue(i);
-      colors[i + 1] = colAttributeData.GetValue(i + 1);
-      colors[i + 2] = colAttributeData.GetValue(i + 2);
+      attributeDataArray[i] = attributeData.GetValue(i);
```

Comment:
I believe I made a mistake here that texture coordinates should not have three vertex data. In which case we can not use 3 for size of per attribute data. It's my bad that I didn't test a model with texture coordinates. This PR LGTM if you could fix this? I think a simple way will be just check if decoderAttr === decoderModule.TEX_COORD then change the size of per vertex data?